app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt

```diff
@@ -173,16 +96,28 @@ class RoutePresenter(
         }
     }
 
-    private fun showUnlockFallback(action: RouteAction.UnlockFallbackDialog) {
-        val manager = activity.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
-        val intent = manager.createConfirmDeviceCredentialIntent(
-            activity.getString(R.string.unlock_fallback_title),
-            activity.getString(R.string.confirm_pattern)
-        )
-        currentFragment.startActivityForResult(intent, action.requestCode)
+    fun showAutofillDialogFragment(dialogFragment: DialogFragment, destination: RouteAction.DialogFragment) {
+        val fragmentManager = activity.supportFragmentManager
+        try {
+            dialogFragment.show(fragmentManager, dialogFragment.javaClass.name)
+            dialogFragment.setupDialog(destination.dialogTitle, destination.dialogSubtitle)
+        } catch (e: IllegalStateException) {
+            log.error("Could not show dialog", e)
+        }
+    }
```

Comment:
This should use the `RoutePresenter.showDialogFragment`, which uses `currentFragment` and `navHostFragmentManager`.